--- flac-1.1.0/src/libFLAC/Makefile.am.orig	2003-01-25 19:25:17.000000000 +0100
+++ flac-1.1.0/src/libFLAC/Makefile.am	2003-12-28 14:55:17.561056584 +0100
@@ -27,10 +27,11 @@
 if FLaC__CPU_IA32
 if FLaC__HAS_NASM
 ARCH_SUBDIRS = ia32
-libFLAC_la_LIBADD = ia32/libFLAC-asm.la
+FLAC_ASM = ia32/libFLAC-asm.la
 endif
 endif
 endif
+libFLAC_la_LIBADD = $(FLAC_ASM) -lm
 
 SUBDIRS = $(ARCH_SUBDIRS) include .
 
--- flac-1.1.0/src/libFLAC++/Makefile.am.orig	2003-01-25 19:25:14.000000000 +0100
+++ flac-1.1.0/src/libFLAC++/Makefile.am	2003-12-28 14:56:37.565894000 +0100
@@ -30,6 +30,7 @@
 
 # see 'http://www.gnu.org/software/libtool/manual.html#Versioning' for numbering convention
 libFLAC___la_LDFLAGS = -version-info 3:2:1
+libFLAC___la_LIBADD = $(top_builddir)/src/libFLAC/libFLAC.la
 
 libFLAC___la_SOURCES = \
 	file_decoder.cpp \
--- flac-1.1.0/src/libOggFLAC/Makefile.am.orig	2003-01-25 19:25:24.000000000 +0100
+++ flac-1.1.0/src/libOggFLAC/Makefile.am	2003-12-28 15:01:26.721935600 +0100
@@ -33,7 +33,7 @@
 	libOggFLAC_static.dsp \
 	libOggFLAC.m4
 
-libOggFLAC_la_LIBADD = @OGG_LIBS@
+libOggFLAC_la_LIBADD = @OGG_LIBS@ $(top_builddir)/src/libFLAC/libFLAC.la
 # see 'http://www.gnu.org/software/libtool/manual.html#Versioning' for numbering convention
 libOggFLAC_la_LDFLAGS = -version-info 1:2:0
 
--- flac-1.1.0/src/libOggFLAC++/Makefile.am.orig	2003-01-25 19:25:21.000000000 +0100
+++ flac-1.1.0/src/libOggFLAC++/Makefile.am	2003-12-28 15:03:41.092508168 +0100
@@ -28,6 +28,7 @@
 	libOggFLAC++_static.dsp \
 	libOggFLAC++.m4
 
+libOggFLAC___la_LIBADD = $(top_builddir)/src/libOggFLAC/libOggFLAC.la
 # see 'http://www.gnu.org/software/libtool/manual.html#Versioning' for numbering convention
 libOggFLAC___la_LDFLAGS = -version-info 0:4:0
 
