diff -Nru flac-1.1.0.orig/configure.in flac-1.1.0/configure.in
--- flac-1.1.0.orig/configure.in	Wed Jan 15 21:56:07 2003
+++ flac-1.1.0/configure.in	Wed Jan 29 18:52:47 2003
@@ -248,9 +248,6 @@
 AC_DEFINE(FLAC__HAS_OGG)
 fi
 
-AM_PATH_XMMS(0.9.5.1, , AC_MSG_WARN([*** XMMS >= 0.9.5.1 not installed - xmms support will not be built]))
-AM_CONDITIONAL(FLaC__HAS_XMMS, test x$XMMS_INPUT_PLUGIN_DIR != x)
-
 dnl check for i18n(internationalization); these are from libiconv/gettext
 AM_ICONV
 AM_LANGINFO_CODESET
diff -Nru flac-1.1.0.orig/src/Makefile.am flac-1.1.0/src/Makefile.am
--- flac-1.1.0.orig/src/Makefile.am	Sat Jan 25 19:14:45 2003
+++ flac-1.1.0/src/Makefile.am	Wed Jan 29 18:53:35 2003
@@ -15,10 +15,6 @@
 #  along with this program; if not, write to the Free Software
 #  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
-if FLaC__HAS_XMMS
-XMMS_DIRS = plugin_xmms
-endif
-
 if FLaC__HAS_OGG
 OGGFLAC_DIRS = libOggFLAC libOggFLAC++
 OGGFLAC_TEST_DIRS = test_libOggFLAC test_libOggFLAC++
@@ -33,7 +29,6 @@
 	metaflac \
 	monkeys_audio_utilities \
 	plugin_common \
-	$(XMMS_DIRS) \
 	plugin_winamp2 \
 	plugin_winamp3 \
 	test_grabbag \
