diff -urN flac-1.2.1/configure.in flac-1.2.1.new/configure.in
--- flac-1.2.1/configure.in	2007-09-13 17:48:42.000000000 +0200
+++ flac-1.2.1.new/configure.in	2009-05-30 14:43:18.867607346 +0200
@@ -29,6 +29,10 @@
 # one that builds only static.  This is because the resulting libtool
 # does not allow us to choose which to build at runtime.
 AM_PROG_LIBTOOL
+
+dnl when using libtool 2.x create libtool early, because it's used in configure
+m4_ifdef([LT_OUTPUT], [LT_OUTPUT])
+
 sed -e 's/^build_old_libs=yes/build_old_libs=no/' libtool > libtool-disable-static
 chmod +x libtool-disable-static
 
