--- flac-1.1.0/src/libFLAC/Makefile.am.orig	2003-01-25 19:25:17.000000000 +0100
+++ flac-1.1.0/src/libFLAC/Makefile.am	2004-12-29 20:15:38.516444056 +0100
@@ -27,10 +27,11 @@
 if FLaC__CPU_IA32
 if FLaC__HAS_NASM
 ARCH_SUBDIRS = ia32
-libFLAC_la_LIBADD = ia32/libFLAC-asm.la
+FLAC_ASM = ia32/libFLAC-asm.la
 endif
 endif
 endif
+libFLAC_la_LIBADD = $(FLAC_ASM) -lm
 
 SUBDIRS = $(ARCH_SUBDIRS) include .
 
--- flac-1.1.0/src/libFLAC++/Makefile.am.orig	2003-01-25 19:25:14.000000000 +0100
+++ flac-1.1.0/src/libFLAC++/Makefile.am	2004-12-29 20:15:38.516444056 +0100
@@ -30,6 +30,7 @@
 
 # see 'http://www.gnu.org/software/libtool/manual.html#Versioning' for numbering convention
 libFLAC___la_LDFLAGS = -version-info 3:2:1
+libFLAC___la_LIBADD = $(top_builddir)/src/libFLAC/libFLAC.la
 
 libFLAC___la_SOURCES = \
 	file_decoder.cpp \
--- flac-1.1.0/src/libOggFLAC/Makefile.am.orig	2003-01-25 19:25:24.000000000 +0100
+++ flac-1.1.0/src/libOggFLAC/Makefile.am	2004-12-29 20:15:38.517443904 +0100
@@ -33,7 +33,7 @@
 	libOggFLAC_static.dsp \
 	libOggFLAC.m4
 
-libOggFLAC_la_LIBADD = @OGG_LIBS@
+libOggFLAC_la_LIBADD = @OGG_LIBS@ $(top_builddir)/src/libFLAC/libFLAC.la
 # see 'http://www.gnu.org/software/libtool/manual.html#Versioning' for numbering convention
 libOggFLAC_la_LDFLAGS = -version-info 1:2:0
 
--- flac-1.1.0/src/libOggFLAC++/Makefile.am.orig	2003-01-25 19:25:21.000000000 +0100
+++ flac-1.1.0/src/libOggFLAC++/Makefile.am	2004-12-29 20:15:38.517443904 +0100
@@ -28,6 +28,7 @@
 	libOggFLAC++_static.dsp \
 	libOggFLAC++.m4
 
+libOggFLAC___la_LIBADD = $(top_builddir)/src/libOggFLAC/libOggFLAC.la
 # see 'http://www.gnu.org/software/libtool/manual.html#Versioning' for numbering convention
 libOggFLAC___la_LDFLAGS = -version-info 0:4:0
 
--- flac-1.1.0/src/share/gain_analysis/Makefile.am.orig	2003-01-25 19:15:37.000000000 +0100
+++ flac-1.1.0/src/share/gain_analysis/Makefile.am	2004-12-29 20:16:49.145706776 +0100
@@ -4,9 +4,9 @@
 
 INCLUDES = -I$(top_srcdir)/include/share
 
-noinst_LIBRARIES = libgain_analysis.a
+noinst_LTLIBRARIES = libgain_analysis.la
 
-libgain_analysis_a_SOURCES = gain_analysis.c
+libgain_analysis_la_SOURCES = gain_analysis.c
 
 EXTRA_DIST = \
 	Makefile.lite \
--- flac-1.1.0/src/share/getopt/Makefile.am.orig	2003-01-25 19:15:33.000000000 +0100
+++ flac-1.1.0/src/share/getopt/Makefile.am	2004-12-29 20:18:42.113533056 +0100
@@ -4,9 +4,9 @@
 
 INCLUDES = -I$(top_srcdir)/include/share
 
-noinst_LIBRARIES = libgetopt.a
+noinst_LTLIBRARIES = libgetopt.la
 
-libgetopt_a_SOURCES = getopt.c getopt1.c
+libgetopt_la_SOURCES = getopt.c getopt1.c
 
 EXTRA_DIST = \
 	Makefile.lite \
--- flac-1.1.0/src/share/grabbag/Makefile.am.orig	2003-01-25 19:15:39.000000000 +0100
+++ flac-1.1.0/src/share/grabbag/Makefile.am	2004-12-29 20:17:22.297666912 +0100
@@ -4,9 +4,9 @@
 
 INCLUDES = -I$(top_srcdir)/include
 
-noinst_LIBRARIES = libgrabbag.a
+noinst_LTLIBRARIES = libgrabbag.la
 
-libgrabbag_a_SOURCES = \
+libgrabbag_la_SOURCES = \
 	cuesheet.c \
 	file.c \
 	replaygain.c \
--- flac-1.1.0/src/share/utf8/Makefile.am.orig	2003-01-25 19:15:36.000000000 +0100
+++ flac-1.1.0/src/share/utf8/Makefile.am	2004-12-29 20:18:11.189234264 +0100
@@ -4,9 +4,9 @@
 
 INCLUDES = -I$(top_srcdir)/include/share
 
-noinst_LIBRARIES = libutf8.a
+noinst_LTLIBRARIES = libutf8.la
 
-libutf8_a_SOURCES = charset.c charset.h iconvert.c utf8.c
+libutf8_la_SOURCES = charset.c charset.h iconvert.c utf8.c
 
 EXTRA_DIST = \
 	Makefile.lite \
--- flac-1.1.0/src/flac/Makefile.am.orig	2003-01-25 19:14:42.000000000 +0100
+++ flac-1.1.0/src/flac/Makefile.am	2004-12-29 20:20:02.112371384 +0100
@@ -41,10 +41,10 @@
 
 flac_LDADD = \
 	$(NEED_OGGFLAC_LIB) \
-	$(top_builddir)/src/share/grabbag/libgrabbag.a \
-	$(top_builddir)/src/share/gain_analysis/libgain_analysis.a \
-	$(top_builddir)/src/share/getopt/libgetopt.a \
-	$(top_builddir)/src/share/utf8/libutf8.a \
+	$(top_builddir)/src/share/grabbag/libgrabbag.la \
+	$(top_builddir)/src/share/gain_analysis/libgain_analysis.la \
+	$(top_builddir)/src/share/getopt/libgetopt.la \
+	$(top_builddir)/src/share/utf8/libutf8.la \
 	$(top_builddir)/src/libFLAC/libFLAC.la \
 	@OGG_LIBS@ \
 	@LIBICONV@ \
--- flac-1.1.0/src/metaflac/Makefile.am.orig	2003-01-25 19:14:53.000000000 +0100
+++ flac-1.1.0/src/metaflac/Makefile.am	2004-12-29 20:20:33.753561192 +0100
@@ -39,10 +39,10 @@
 metaflac_LDFLAGS = 
 
 metaflac_LDADD = \
-	$(top_builddir)/src/share/grabbag/libgrabbag.a \
-	$(top_builddir)/src/share/gain_analysis/libgain_analysis.a \
-	$(top_builddir)/src/share/getopt/libgetopt.a \
-	$(top_builddir)/src/share/utf8/libutf8.a \
+	$(top_builddir)/src/share/grabbag/libgrabbag.la \
+	$(top_builddir)/src/share/gain_analysis/libgain_analysis.la \
+	$(top_builddir)/src/share/getopt/libgetopt.la \
+	$(top_builddir)/src/share/utf8/libutf8.la \
 	$(top_builddir)/src/libFLAC/libFLAC.la \
 	@LIBICONV@ \
 	-lm
--- flac-1.1.0/src/plugin_common/Makefile.am.orig	2003-01-25 19:15:48.000000000 +0100
+++ flac-1.1.0/src/plugin_common/Makefile.am	2004-12-29 20:19:20.237737296 +0100
@@ -6,7 +6,7 @@
 
 INCLUDES = -I$(top_srcdir)/include @ID3LIB_CFLAGS@
 
-noinst_LIBRARIES = libplugin_common.a
+noinst_LTLIBRARIES = libplugin_common.la
 
 noinst_HEADERS = \
 	all.h \
@@ -20,7 +20,7 @@
 	replaygain_synthesis.h \
 	vorbiscomment.h
 
-libplugin_common_a_SOURCES = \
+libplugin_common_la_SOURCES = \
 	canonical_tag.c \
 	charset.c \
 	dither.c \
--- flac-1.1.0/src/plugin_xmms/Makefile.am.orig	2003-01-14 02:33:03.000000000 +0100
+++ flac-1.1.0/src/plugin_xmms/Makefile.am	2004-12-29 20:21:34.061393008 +0100
@@ -57,12 +57,11 @@
 #   http://lists.freshrpms.net/pipermail/rpm-list/2002-April/000746.html
 # the workaround is the extra '-L$(top_builddir)/src/libFLAC/.libs'
 libxmms_flac_la_LIBADD = \
-	$(top_builddir)/src/plugin_common/libplugin_common.a \
-	$(top_builddir)/src/share/grabbag/libgrabbag.a \
-	$(top_builddir)/src/share/gain_analysis/libgain_analysis.a \
-	$(top_builddir)/src/share/utf8/libutf8.a \
+	$(top_builddir)/src/plugin_common/libplugin_common.la \
+	$(top_builddir)/src/share/grabbag/libgrabbag.la \
+	$(top_builddir)/src/share/gain_analysis/libgain_analysis.la \
+	$(top_builddir)/src/share/utf8/libutf8.la \
 	$(top_builddir)/src/libFLAC/libFLAC.la \
-	-L$(top_builddir)/src/libFLAC/.libs \
 	@XMMS_LIBS@ \
 	@ID3LIB_LIBS@
 libxmms_flac_la_LDFLAGS = -module -avoid-version
--- flac-1.1.0/src/test_grabbag/cuesheet/Makefile.am.orig	2003-01-25 19:15:58.000000000 +0100
+++ flac-1.1.0/src/test_grabbag/cuesheet/Makefile.am	2004-12-29 20:22:18.241676584 +0100
@@ -25,8 +25,8 @@
 test_cuesheet_SOURCES = \
 	main.c
 test_cuesheet_LDADD = \
-	$(top_builddir)/src/share/grabbag/libgrabbag.a \
-	$(top_builddir)/src/share/gain_analysis/libgain_analysis.a \
+	$(top_builddir)/src/share/grabbag/libgrabbag.la \
+	$(top_builddir)/src/share/gain_analysis/libgain_analysis.la \
 	$(top_builddir)/src/libFLAC/libFLAC.la \
 	-lm
 
--- flac-1.1.0/src/test_libFLAC/Makefile.am.orig	2003-01-25 19:15:40.000000000 +0100
+++ flac-1.1.0/src/test_libFLAC/Makefile.am	2004-12-29 20:22:36.583888144 +0100
@@ -24,8 +24,8 @@
 
 noinst_PROGRAMS = test_libFLAC
 test_libFLAC_LDADD = \
-	$(top_builddir)/src/share/grabbag/libgrabbag.a \
-	$(top_builddir)/src/share/gain_analysis/libgain_analysis.a \
+	$(top_builddir)/src/share/grabbag/libgrabbag.la \
+	$(top_builddir)/src/share/gain_analysis/libgain_analysis.la \
 	$(top_builddir)/src/libFLAC/libFLAC.la \
 	-lm
 test_libFLAC_SOURCES = \
--- flac-1.1.0/src/test_libFLAC++/Makefile.am.orig	2003-01-25 19:15:42.000000000 +0100
+++ flac-1.1.0/src/test_libFLAC++/Makefile.am	2004-12-29 20:22:55.074077208 +0100
@@ -24,8 +24,8 @@
 
 noinst_PROGRAMS = test_libFLAC++
 test_libFLAC___LDADD = \
-	$(top_builddir)/src/share/grabbag/libgrabbag.a \
-	$(top_builddir)/src/share/gain_analysis/libgain_analysis.a \
+	$(top_builddir)/src/share/grabbag/libgrabbag.la \
+	$(top_builddir)/src/share/gain_analysis/libgain_analysis.la \
 	$(top_builddir)/src/libFLAC++/libFLAC++.la \
 	$(top_builddir)/src/libFLAC/libFLAC.la \
 	-lm
--- flac-1.1.0/src/test_libOggFLAC/Makefile.am.orig	2003-01-25 19:15:45.000000000 +0100
+++ flac-1.1.0/src/test_libOggFLAC/Makefile.am	2004-12-29 20:23:13.633255784 +0100
@@ -24,8 +24,8 @@
 
 noinst_PROGRAMS = test_libOggFLAC
 test_libOggFLAC_LDADD = \
-	$(top_builddir)/src/share/grabbag/libgrabbag.a \
-	$(top_builddir)/src/share/gain_analysis/libgain_analysis.a \
+	$(top_builddir)/src/share/grabbag/libgrabbag.la \
+	$(top_builddir)/src/share/gain_analysis/libgain_analysis.la \
 	$(top_builddir)/src/libOggFLAC/libOggFLAC.la \
 	$(top_builddir)/src/libFLAC/libFLAC.la \
 	@OGG_LIBS@ \
--- flac-1.1.0/src/test_libOggFLAC++/Makefile.am.orig	2003-01-25 19:15:46.000000000 +0100
+++ flac-1.1.0/src/test_libOggFLAC++/Makefile.am	2004-12-29 20:23:30.614674216 +0100
@@ -24,8 +24,8 @@
 
 noinst_PROGRAMS = test_libOggFLAC++
 test_libOggFLAC___LDADD = \
-	$(top_builddir)/src/share/grabbag/libgrabbag.a \
-	$(top_builddir)/src/share/gain_analysis/libgain_analysis.a \
+	$(top_builddir)/src/share/grabbag/libgrabbag.la \
+	$(top_builddir)/src/share/gain_analysis/libgain_analysis.la \
 	$(top_builddir)/src/libOggFLAC++/libOggFLAC++.la \
 	$(top_builddir)/src/libOggFLAC/libOggFLAC.la \
 	$(top_builddir)/src/libFLAC/libFLAC.la \
